### ISSUE TITLE
Bad JSONFormat rendering of nested records

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13138]
The generation of formatted JSON results from `result.formatJSON(new JSONFormat().format(true))` is off. E.g. this:

```java
Result<Record2<String, Record2<String, String>>> result =
    create.newResult(FIELD_NAME1, rowField(row(FIELD_NAME1, FIELD_NAME2)));

result.add(record("a", record("b", "c")));
result.add(record("x", record("y", "z")));
```

Should produce something like this:

```
[
  [
    "a",
    [
      "b",
      "c"
    ]
  ],
  [
    "x",
    [
      "y",
      "z"
    ]
  ]
]
```

But instead produces

```
[
  [
    "a",
        [
      "b",
      "c"
    ]
  ],
  [
    "x",
        [
      "y",
      "z"
    ]
  ]
]
```

Looks like the indentations are off.

**Comment by lukaseder** [comment]
Both record layouts are affected: arrays and objects.

**Comment by lukaseder** [comment]
The implementation of https://github.com/jOOQ/jOOQ/issues/7782 is wrong. Our tests seem to have covered only unformatted output, which is correct.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0, 3.16.5 (https://github.com/jOOQ/jOOQ/issues/13140), and 3.15.9 (https://github.com/jOOQ/jOOQ/issues/13141)
